src/loaders/FileLoader.js

```diff
@@ -55,225 +55,190 @@ class FileLoader extends Loader {
 
 		}
 
-		// Check for data: URI
-		const dataUriRegex = /^data:(.*?)(;base64)?,(.*)$/;
-		const dataUriRegexResult = url.match( dataUriRegex );
-		let request;
+		// Initialise array for duplicate requests
+		loading[ url ] = [];
+
+		loading[ url ].push( {
+			onLoad: onLoad,
+			onProgress: onProgress,
+			onError: onError,
+		} );
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType
+		// if (this.responseType !== undefined) {
+		//   request.responseType = this.responseType;
+		// }
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/overrideMimeType
+		// if (request.overrideMimeType) {
+		//   request.overrideMimeType(
+		//     this.mimeType !== undefined ? this.mimeType : "text/plain",
+		//   );
+		// }
+
+		// TODO-DefinitelyMaybe: Confirm if Safari can handle Data URIs through fetch
+		// create request
+		const req = new Request( url, {
+			headers: new Headers( this.requestHeader ),
+			credentials: this.withCredentials ? 'include' : 'same-origin',
+			// TODO-DefinitelyMaybe: Could add an abort controller?
+			// signal: this.attachSignal ? this.signal : undefined;
+		} );
+
+		// start the fetch
+		fetch( req )
+			.then( response => {
+
+				if ( response.status === 200 || response.status === 0 ) {
 
-		// Safari can not handle Data URIs through XMLHttpRequest so process manually
-		if ( dataUriRegexResult ) {
-
-			const mimeType = dataUriRegexResult[ 1 ];
-			const isBase64 = !! dataUriRegexResult[ 2 ];
-
-			let data = dataUriRegexResult[ 3 ];
-			data = decodeURIComponent( data );
-
-			if ( isBase64 ) data = atob( data );
-
-			try {
-
-				let response;
-				const responseType = ( this.responseType || '' ).toLowerCase();
-
-				switch ( responseType ) {
-
-					case 'arraybuffer':
-					case 'blob':
-
-						const view = new Uint8Array( data.length );
+					// Some browsers return HTTP Status 0 when using non-http protocol
+					// e.g. 'file://' or 'data://'. Handle as success.
 
-						for ( let i = 0; i < data.length; i ++ ) {
+					if ( response.status === 0 ) {
 
-							view[ i ] = data.charCodeAt( i );
+						console.warn( 'THREE.FileLoader: HTTP Status 0 received.' );
 
-						}
+					}
 
-						if ( responseType === 'blob' ) {
+					const callbacks = loading[ url ];
+					const reader = response.body.getReader();
+					let loaded = 0;
+					const total = parseInt( response.headers.get( 'Content-Length' ) );
```

Comment:
I don't believe there's any guarantee that the 'Content-Length' header will be available from a request (see #15584). If it's not available then `lengthComputable` should be false and `total` set to 0.

Reply:
good point. have updated.